src/arch/e2k/context_switch.S

```diff
@@ -0,0 +1,89 @@
+/**
+ * @file
+ * @brief Context switch for E2K
+ *
+ * @date 18.03.2019
+ * @author Alexander Kalmuk
+ */
+
+#include <hal/context.h>
+#include <e2k_api.h>
+
+.section ".text", "ax"
+
+/* TODO Save and restore Global Registers. */
+
+.global $context_switch
```

Comment:
may be create macro C_ENTRY?

Reply:
Thanks, I defined such macro. Probably, it should be placed in the common asm/linkage.h in the future.